### PR TITLE
Update to glib 0.14

### PR DESCRIPTION
**Pull request by cgwalters**
An intimidating spam of compiler errors at the start, but the
biggest was handling the new convention of `ostree_sys::` => `ffi::`.

This will require a semver bump of course.

---

### CODE REVIEW

**Comment by travier** [comment]
Superficial LGTM. Looks like a lot of files are auto-generated. Which one aren't so we can focus on those?

**Comment by cgwalters** [comment]
Everything in `src/auto` is generated.  Pushed a fix for that.

What isn't as easily fixed is that `cfg-expr` [hard requires the new pattern match bits](https://github.com/EmbarkStudios/cfg-expr/commit/7f856d7dec36943fbc0cc1f8310750d59b434137) in Rust 1.53 released a mere 10 days ago.  And that dep flows from `glib-sys` :arrow_right: `system-deps`.

File: src/lib.rs [hunk 1]
```diff
@@ -9,12 +9,9 @@
 extern crate gio_sys;
 extern crate glib_sys;
 extern crate gobject_sys;
-extern crate ostree_sys;
```
Comment (lucab) [inline src/lib.rs:12]: If you are enjoying spring cleanups, I think we can directly drop all the `extern crate` entries here as the PR moves to 2018 edition.

Comment (cgwalters) [reply src/lib.rs:12]: Yep, done!

**Comment by cgwalters** [comment]
OK cool, passing CI now.

**Review by lucab — APPROVED** [review]

**Pull request merged**